src/Raven.Server/Documents/Sharding/Handlers/ShardedSubscriptionsHandler.cs

```diff
@@ -32,13 +32,25 @@ public async Task Update()
             using (var processor = new ShardedSubscriptionsHandlerProcessorForPostSubscription(this))
                 await processor.ExecuteAsync();
         }
-        
+
+        internal void ThrowOnRevisionsIfNeeded(bool subRevisions)
+        {
+            if (subRevisions)
+            {
+                // RavenDB-18881
+                throw new NotImplementedException(@"Revisions subscription is not supported for sharding database.");
```

Comment:
We have `NotSupportedInShardingException` IIRC

Reply:
yes